File: Sliding window/LET_no209_Minimum Size Subarray Sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        left = 0
        res = float('inf')
        for right in range(len(nums)):
            target -= nums[right]
            while left <= right and target <= 0:
                res = min(res, right - left + 1)
                target += nums[left]
                left += 1
            
        return res if res != float('inf') else 0
                

File: Sliding window/LET_no209_Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0, n = nums.size(), res = n + 1;
        for (int right = 0; right < n; right++) {
            target -= nums[right];
            while (left <= right && target <= 0) {
                res = min(res, right - left + 1);
                target += nums[left++];
            }
        }

        return (res == n + 1 ? 0 : res);
    }
};